Parse/Internal/LocalDataStore/SQLite/PFSQLiteDatabase.m

```diff
@@ -225,9 +225,14 @@ - (BFTask *)executeCachedQueryAsync:(NSString *)sql withArgumentsInArray:(NSArra
     }];
 }
 
-- (BFTask *)executeQueryAsync:(NSString *)sql withArgumentsInArray:(NSArray *)args {
+- (BFTask *)executeQueryAsync:(NSString *)query withArgumentsInArray:(nullable NSArray *)args block:(PFSQLiteDatabaseQueryBlock)block {
     return [BFTask taskFromExecutor:_databaseExecutor withBlock:^id {
-        return [self _executeQueryAsync:sql withArgumentsInArray:args cachingEnabled:NO];
+        BFTask *task = [self _executeQueryAsync:query withArgumentsInArray:args cachingEnabled:NO];
+        return [task continueImmediatelyWithSuccessBlock:^id(BFTask PF_GENERIC(PFSQLiteDatabaseResult *)*task) {
+            id result = block(task.result);
```

Comment:
If `block` throws during execution, we end up with statements not being properly closed. If we ever turn on query caching, we get into a really bad state here.